feat: add validation for `preserveEntrySignatures` with `includeDependenciesRecursively` option

hyf0 opened the pull request:
Closes #6449

Two varificationd:

When `includeDependenciesRecursively` is set to false:

- if `preserveEntrySignatures` is unset, we will set it `allow-extension` with warning.
- if `preserveEntrySignatures` is set to `strcit` etc, we will throw an error to interrupt the build.

hyf0 commented:
* **#6468** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6468?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6468?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6468).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

IWANABETHATGUY commented on crates/rolldown/tests/rolldown/warnings/invalid_option/include_dependencies_recursively_implicit_preserve_entry_signatures/artifacts.snap:
````diff
@@ -0,0 +1,44 @@
+```text
+[INVALID_OPTION] Warning: `preserveEntrySignatures: 'allow-extension'` is set implicitly by Rolldown
+
+- `advancedChunks.includeDependenciesRecursively = false` requires `preserveEntrySignatures` to be either `false` or 'allow-extension'
````
probably just `advancedChunks.includeDependenciesRecursively: false` , ?

hyf0 replied:
```
- `advancedChunks.includeDependenciesRecursively = false` requires `preserveEntrySignatures` to be either `false` or 'allow-extension'
```
explains the reason of title.

I think user might be confused about a standalone

```
`advancedChunks.includeDependenciesRecursively: false`
```

IWANABETHATGUY commented on crates/rolldown_error/src/build_diagnostic/events/invalid_option.rs:
```diff
@@ -40,6 +42,29 @@ impl BuildEvent for InvalidOption {
+            "",
+            "To fix:",
+            "",
+            "- Set `preserveEntrySignatures` either to `false` or 'allow-extension' in your config",
```
We should also inform the user that we will assign the default value `allow-extension` if the user does not provide an explicit value.

hyf0 replied:
This is an error. We won't assign the default value in this case.

IWANABETHATGUY replied:
<img width="1225" height="466" alt="Image" src="https://github.com/user-attachments/assets/35946798-f480-4de6-9f88-d4192cad8dee" />

hyf0 replied:
<img width="1154" height="1432" alt="image" src="https://github.com/user-attachments/assets/d5c4b7e5-a2bc-46c2-96b6-b66ab852f963" />

It's this one and I found the call of `with_severty_warning`. I'll fix it. But still, it's an error.

`with_severty_warning` is fixed.

hyf0 replied:
I added more description in the PR.

hyf0 committed:
feat: add validation for `preserveEntrySignatures` with `includeDependenciesRecursively` option (#6468)

Closes #6449

Two varificationd:

When `includeDependenciesRecursively` is set to false:

- if `preserveEntrySignatures` is unset, we will set it `allow-extension` with warning.
- if `preserveEntrySignatures` is set to `strcit` etc, we will throw an error to interrupt the build.
